apps/arweave/src/ar_data_discovery.erl

```diff
@@ -67,6 +70,32 @@ get_bucket_peers(Bucket, Cursor, Peers) ->
 			PickedPeers
 	end.
 
+%% @doc Return the list of ?QUERY_BEST_PEERS_COUNT peers who have at least one byte of
+%% data synced in the given footprint bucket of size ?NETWORK_FOOTPRINT_BUCKET_SIZE.
+%% 80% of the peers are chosen from the 20% of peers with the biggest share
+%% in the given bucket.
+get_footprint_bucket_peers(Bucket) ->
+	case ets:member(ar_peers, block_connections) of
+		true ->
+			[];
+		false ->
+			get_footprint_bucket_peers(Bucket, {Bucket, 0, no_peer}, [])
+	end.
+
+get_footprint_bucket_peers(Bucket, Cursor, Peers) ->
+	case ets:next(ar_data_discovery_footprint_buckets, Cursor) of
+		'$end_of_table' ->
+			UniquePeers = sets:to_list(sets:from_list(Peers)),
+			PickedPeers = pick_peers(UniquePeers, ?QUERY_BEST_PEERS_COUNT),
+			PickedPeers;
+		{Bucket, _Share, Peer} = Key ->
+			get_footprint_bucket_peers(Bucket, Key, [Peer | Peers]);
```

Comment:
This logic will cause it to keep advancing through the ETS table accumulating Peers until it hits a entry from a new bucket, right? At which point the `_` branch is entered and the peers are picked

Reply:
Yep